lua/presence/init.lua

```diff
@@ -502,16 +502,21 @@ function Presence:update_for_buffer(buffer, should_debounce)
         state = status_text,
         assets = assets,
         timestamps = {
-            start = activity_set_at,
+            start = relative_activity_set_at,
         },
     }
 
-    self.log:debug(string.format("Getting project name for %s...", parent_dirpath))
+    if not self.options.enable_line_number then
+        self.log:debug("Getting line number for current buffer...")
+    else
+        self.log:debug(string.format("Getting project name for %s...", parent_dirpath))
+    end
+
     local workspace_text = self.options.workspace_text
     local project_name, project_path = self:get_project_name(parent_dirpath)
 
-    -- Include project details if available
-    if project_name then
+    -- Include project details if available and if the user hasn't set the enable_line_number option
+    if project_name and self.options.enable_line_number == 0 then
```

Comment:
Hmm so it looks like currently the plugin tries to get the project name even though it won't be used when the line number option is enabled. We might want to refactor these `if`/`elseif`/`else` statements to something like this:
```lua
if self.options.enable_line_number then
    self.log:debug("Getting line number for current buffer...")
    -- your code to get line number info and set activity
else
    self.log:debug(string.format("Getting project name for %s...", parent_dirpath))

    local workspace_text = self.options.workspace_text
    local project_name, project_path = self:get_project_name(parent_dirpath)

    if project_name then
        -- existing "Detected project: %s" code
    else
        -- existing "No project detected" code
    end
end
```
That way we only call `self:get_project_name` when necessary. Also note that the condition checks `self.options.enable_line_number` as a `boolean` and not a `0` or `1` number, since the `0` and `1` values from global vim variables get coalesced into `true` and `false`.